addons/sale_timesheet/models/sale_order.py

```diff
@@ -169,6 +170,21 @@ def _compute_product_updatable(self):
             else:
                 super(SaleOrderLine, line)._compute_product_updatable()
 
+    def _auto_init(self):
+        """
+        Create column to stop ORM from computing it himself (too slow)
+        """
+        if not column_exists(self.env.cr, 'sale_order_line', 'is_service'):
+            create_column(self.env.cr, 'sale_order_line', 'is_service', 'bool')
+            self.env.cr.execute("""
+                UPDATE sale_order_line line
+                SET is_service = (pt.type = 'service')
+                FROM product_product pp
+                LEFT JOIN product_template pt ON pt.id = pp.product_tmpl_id
+                WHERE pp.id = line.product_id
+            """)
```

Comment:
This will still call `_init_column` and perform a useless `UPDATE` query.

I think this part should be done in an override to `_init_column` if not for the reason mentioned above, then at least for consistency's sake. We should make this the convention when making these kind of local performance patches as it's what we've already done in the past.

Reply:
Do you mean to move UPDATE to ``_init_column``? I tried and it doesn't work, because the method is not called if column already exists

https://github.com/odoo/odoo/blob/5d5b6eb2f9335b4fa6e2709e99b0528862472a2e/odoo/fields.py#L976-L979

Reply:
Yes that's what I meant, are you sure about `_init_column` not being called? It should work if installing/upgrading(?) `sale_timesheet`, the column has to exist because `_init_column` does not create columns, it only sets the initial values

Reply:
Yes, ``_inti_column`` is not called and you can see it from the code of update_db, update_db_notnull methods.

I meant that column should not exist before calling of ``update_db`` method, otherwise ``_inti_column`` is not called

https://github.com/odoo/odoo/blob/5d5b6eb2f9335b4fa6e2709e99b0528862472a2e/odoo/fields.py#L903-L923

Reply:
@yelizariev you're right, then that means that #67677 is potentially wrong, no? what happens if you only keep the `_init_column`? will the installation still timeout? I don't remember exactly how computes work in 12.0 and I know from reading the code that the field will mark the records to be recomputed, but does it actually happen?

If so you can keep your previous version with the update in the `_auto_init` and we'll merge that, sorry 😜

Reply:
> what happens if you only keep the `_init_column`? will the installation still timeout?

In this case we execute UPDATE, but then the field is recomputed via _compute_* methods anyway